ee/tabby-schema/src/schema/mod.rs

```diff
@@ -564,6 +568,34 @@ impl Query {
         )
         .await
     }
+
+    async fn custom_web_documents(
+        ctx: &Context,
+        after: Option<String>,
+        before: Option<String>,
+        first: Option<i32>,
+        last: Option<i32>,
+    ) -> Result<Connection<WebDocument>> {
+        query_async(
+            after,
+            before,
+            first,
+            last,
+            |after, before, first, last| async move {
+                ctx.locator
+                    .web_documents()
+                    .list_custom_web_documents(after, before, first, last)
+                    .await
+            },
+        )
+        .await
+    }
+    async fn preset_web_documents(ctx: &Context, active: bool) -> Result<Vec<WebDocument>> {
```

Comment:
```suggestion
    async fn preset_web_documents(ctx: &Context, active: bool) -> Result<Vec<PresetWebDocument>> {
```

consider use pagination if needed

Reply:
I have change it